diff --git a/supportApp/hdf5Src/H5T.c b/supportApp/hdf5Src/H5T.c
index 194fa87..f21c99c 100644
--- a/supportApp/hdf5Src/H5T.c
+++ b/supportApp/hdf5Src/H5T.c
@@ -927,6 +927,8 @@ H5T__init_package(void)
      *------------------------------------------------------------
      */
 
+#ifndef H5_HAVE_VXWORKS
+/* These functions are not working on vxWorks, need to figure out why */
     /* Little-endian 32-bit UNIX time_t */
     H5T_INIT_TYPE(TIME, H5T_UNIX_D32LE_g, COPY, std_u32le, NOSET, -)
 
@@ -938,7 +940,7 @@ H5T__init_package(void)
 
     /* Big-endian 64-bit UNIX time_t */
     H5T_INIT_TYPE(TIME, H5T_UNIX_D64BE_g, COPY, std_u64be, NOSET, -)
-
+#endif
 
     /* Indicate that the types that are created from here down are allocated
      * H5FL_ALLOC(), not copied with H5T_copy()
