diff --git a/supportApp/hdf5Src/H5FDstdio.c b/supportApp/hdf5Src/H5FDstdio.c
index 13ccd5c..05c1312 100644
--- a/supportApp/hdf5Src/H5FDstdio.c
+++ b/supportApp/hdf5Src/H5FDstdio.c
@@ -124,9 +124,21 @@ typedef struct H5FD_stdio_t {
     #define file_offset_t   __int64
     #define file_ftruncate  _chsize_s   /* Supported in VS 2005 or newer */
     #define file_ftell      _ftelli64
+#else
+    #define file_fseek      fseeko64
+    #define file_offset_t   __int64
+    #define file_ftruncate  _chsize_s   /* Supported in VS 2005 or newer */
+    #define file_ftell      ftello64
 #endif /* H5_HAVE_MINGW */
 #endif /* H5_HAVE_WIN32_API */
 
+#ifdef H5_HAVE_VXWORKS
+  #define file_fseek      fseek
+  #define file_offset_t   off_t
+  #define file_ftruncate  vxWorks_ftruncate
+  #define file_ftell      ftell
+#endif
+
 /* If these functions weren't re-defined for Windows, give them
  * more platform-independent names.
  */
